### ISSUE TITLE
Bug: Getting error when ripping from deeer

### ISSUE THREAD

**Issue by zeusoryx** [nathom/streamrip#124]
## Bug Report

**Current Behavior**
When the below command is used for the second time it produces an error and when used again produces the same error. FLACS from deezer are no longer being downloaded. 

**Command used:**

```bash
rip -u rip -u https://www.deezer.com/en/album/132051252
```

**Expected behavior**
Album cver art and .mp3 file was to be downloaded
**`streamrip` Configuration File (find using `rip config -o`):**

```toml
# [downloads]
# Folder where tracks are downloaded to
folder = ""
# Put Qobuz albums in a 'Qobuz' folder, Tidal albums in 'Tidal' etc.
source_subdirectories = false
# Download (and convert) tracks all at once, instead of sequentially. 
# If you are converting the tracks, or have fast internet, this will 
# substantially improve processing speed.
concurrent = false

[qobuz]
# 1: 320kbps MP3, 2: 16/44.1, 3: 24/<=96, 4: 24/>=96
quality = 3
# This will download booklet pdfs that are included with some albums
download_booklets = true

email = ""
# This is an md5 hash of the plaintext password
password = ""
# Do not change
app_id = ""
# Do not change
secrets = []

[tidal]
# 0: 256kbps AAC, 1: 320kbps AAC, 2: 16/44.1 "HiFi" FLAC, 3: 24/44.1 "MQA" FLAC
quality = 3
# This will download videos included in Video Albums.
download_videos = true

# Do not change any of the fields below
user_id = ""
country_code = ""
access_token = ""
refresh_token = ""
# Tokens last 1 week after refresh. This is the Unix timestamp of the expiration
# time. If you haven't used streamrip in more than a week, you may have to log
# in again using `rip config --tidal`
token_expiry = ""

# Doesn't require login
[deezer]
# 0, 1, or 2
quality = 1

[soundcloud]
# Only 0 is available for now
quality = 0

[youtube]
# Only 0 is available for now
quality = 0
# Download the video along with the audio
download_videos = false
# The path to download the videos to
video_downloads_folder = ""

# This stores a list of item IDs so that repeats are not downloaded.
[database.downloads]
enabled = true
path = ""

# If a download fails, the item ID is stored here. Then, `rip repair` can be
# called to retry the downloads
[database.failed_downloads]
enabled = true
path = ""

# Convert tracks to a codec after downloading them.
[conversion]
enabled = false
# FLAC, ALAC, OPUS, MP3, VORBIS, or AAC
codec = "ALAC"
# In Hz. Tracks are downsampled if their sampling rate is greater than this. 
# Value of 48000 is recommended to maximize quality and minimize space
sampling_rate = 48000
# Only 16 and 24 are available. It is only applied when the bit depth is higher
# than this value.
bit_depth = 24

# Filter a Qobuz artist's discography. Set to 'true' to turn on a filter.
[filters]
# Remove Collectors Editions, live recordings, etc.
extras = false
# Picks the highest quality out of albums with identical titles.
repeats = false
# Remove EPs and Singles
non_albums = false
# Remove albums whose artist is not the one requested
features = false
# Skip non studio albums
non_studio_albums = false
# Only download remastered albums
non_remaster = false

[artwork]
# Write the image to the audio file
embed = true
# The size of the artwork to embed. Options: thumbnail, small, large, original.
# "original" images can be up to 30MB, and may fail embedding. 
# Using "large" is recommended.
size = "large"
# Both of these options limit the size of the embedded artwork. If their values
# are larger than the actual dimensions of the image, they will be ignored.
max_width = 999999
max_height = 999999
# Save the cover image at the highest quality as a seperate jpg file
keep_hires_cover = true

[metadata]
# Sets the value of the 'ALBUM' field in the metadata to the playlist's name. 
# This is useful if your music library software organizes tracks based on album name.
set_playlist_to_album = true
# Replaces the original track's tracknumber with it's position in the playlist
new_playlist_tracknumbers = true

# Changes the folder and file names generated by streamrip.
[filepaths]
# Create folders for single tracks within the downloads directory using the folder_format
# template
add_singles_to_folder = false
# Available keys: "albumartist", "title", "year", "bit_depth", "sampling_rate",
# "container", and "albumcomposer"
folder_format = "{albumartist} - {title} ({year}) [{container}] [{bit_depth}B-{sampling_rate}kHz]"
# Available keys: "tracknumber", "artist", "albumartist", "composer", "title",
# and "albumcomposer"
track_format = "{tracknumber}. {artist} - {title}"

# Last.fm playlists are downloaded by searching for the titles of the tracks
[lastfm]
# The source on which to search for the tracks.
source = "qobuz"
# If no results were found with the primary source, the item is searched for 
# on this one.
fallback_source = "deezer"

[misc]
# Check whether a newer version of streamrip is available when starting up
check_for_updates = true

# Metadata to identify this config file. Do not change.
version = "0.7.1"
# REMOVE YOUR CREDENTIALS

```

**Environment**

- `streamrip` version(s): [v0.7.1]
- Python version: latest [3.9.6]
- OS: [Windows 10]

**Traceback**
```bash
# Paste the text that Python dumped on the error here
streamrip is up-to-date!
Traceback (most recent call last):
  File "c:\users\user\appdata\local\programs\python\python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\user\appdata\local\programs\python\python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\user\AppData\Local\Programs\Python\Python39\Scripts\rip.exe\__main__.py", line 7, in <module>
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\rip\cli.py", line 477, in main
    cli(obj={})
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 1646, in invoke
    super().invoke(ctx)
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\click\decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\rip\cli.py", line 121, in cli
    core.download()
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\rip\core.py", line 300, in download
    item.download(**arguments)
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\streamrip\media.py", line 984, in download
    self._prepare_download(**kwargs)
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\streamrip\media.py", line 1319, in _prepare_download
    self.folder = self._get_formatted_folder(
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\streamrip\media.py", line 1485, in _get_formatted_folder
    formatted_folder = clean_format(self.folder_format, self._get_formatter())
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\streamrip\media.py", line 1448, in _get_formatter
    stats = tuple(
  File "c:\users\user\appdata\local\programs\python\python39\lib\site-packages\streamrip\media.py", line 1449, in <genexpr>
    min(bd, sr)
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

**Comment by nathom** [comment]
Can you run the command with the `--debug` option and paste the contents here? Thanks.

**Comment by Kumar-Saksham** [comment]
Hey @nathom seems like the value of **sr** in [line 1449](https://github.com/nathom/streamrip/blob/37e2a7e8c1926cd6762cdd6a6ad2bddf4b4837ee/streamrip/media.py#L1449) is getting a value of None.

As a quick fix I added a check for None values, and it resolved the error for me. 
```python
stats = tuple(
    (bd if sr is None else min(bd, sr)) 
    for (bd, sr) in zip(
        (self.meta.bit_depth, self.meta.sampling_rate),
        get_stats_from_quality(self.quality)
    )
)
```

**Comment by foxmajik** [comment]
Same problem is happening when downloading with Qobuz: URL gets a value of none.

Can you please merge the provided fix?

 ---
DEBUG:streamrip:Downloadable URL found: None
Panic: 'url' dl_info = {'track_id': 16647642, 'duration': 233, 'restrictions': [{'code': 'TrackRestrictedByPurchaseCredentials'}, {'code': 'SampleRestrictedByRightHolders'}], 'sampling_rate': 44.1, 'bit_depth': 16}
Traceback (most recent call last):
  File "c:\python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Python39\Scripts\rip.exe\__main__.py", line 7, in <module>
  File "c:\python39\lib\site-packages\rip\cli.py", line 477, in main
    cli(obj={})
  File "c:\python39\lib\site-packages\click\core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "c:\python39\lib\site-packages\click\core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "c:\python39\lib\site-packages\click\core.py", line 1646, in invoke
    super().invoke(ctx)
  File "c:\python39\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\python39\lib\site-packages\click\core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "c:\python39\lib\site-packages\click\decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "c:\python39\lib\site-packages\rip\cli.py", line 121, in cli
    core.download()
  File "c:\python39\lib\site-packages\rip\core.py", line 300, in download
    item.download(**arguments)
  File "c:\python39\lib\site-packages\streamrip\media.py", line 1028, in download
    target(item, **kwargs)
  File "c:\python39\lib\site-packages\streamrip\media.py", line 1692, in _download_item
    item.download(**kwargs)
  File "c:\python39\lib\site-packages\streamrip\media.py", line 295, in download
    tqdm_download(download_url, self.path, desc=self._progress_desc)
UnboundLocalError: local variable 'download_url' referenced before assignment

**Comment by nathom** [comment]
This has been fixed in `v1.0`
